sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,239 @@
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> power_relators = {x1**2: 1}
+    >>> word = x1**7
+    >>> _reduce_word(word, power_relators)
+    x1
+
+    """
+    group = word.group
+    dividend = word.array_form[0][1]
+    rem = dividend
+    for w, v in power_relators.items():
+        divisor = w.array_form[0][1]
+        if w.letter_form_elm[0] == word.letter_form_elm[0] and dividend % divisor < rem:
+            rem = dividend % divisor
+            quo = dividend // divisor
+            value = power_relators[w]
+    word = (value**quo)*(word.letter_form_elm[0]**rem)
+    return group.dtype(word)
+
+def _conjugate_word(word, conjugate_relators):
+    """
+    Returns the reduced word with the help of
+    conjugate relators.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.pc_groups import _conjugate_word
+    >>> from sympy.combinatorics.free_groups import free_group
+
+    Example 8.7 Pg. 282 from Handbook
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> word = x2**2*x1
+    >>> conjugate_relators = {x1*x2*x1**-1: x2**-1, x1**-1*x2*x1: x2**-1}
+    >>> _conjugate_word(word, conjugate_relators)
+    x1*x2**-2
+
+    """
+    exp = word.array_form[0][1]
+    w = (word[len(word)-1].inverse()*word.letter_form_elm[0]*word.subword(exp, len(word)))
+    if w in conjugate_relators:
+        word = word[len(word)-1]*(conjugate_relators[w]**exp)
+    else:
+        loop = word.array_form[len(word.array_form)-1][1]
+        if loop%2 == 0:
+            word = word[len(word)-1]**loop*word.subword(0, len(word)-loop)
+        else:
+            low, high = abs(word.array_form[0][1]), abs(word.array_form[1][1])
+            high = high + low
+            print(word.subword(low, high))
+            word = word.eliminate_word(word.subword(low, high), word.subword(low, high).inverse())
+            word = word[len(word)-1]**loop*word.subword(0, len(word)-loop)
+    return word
+
+def _index(word, w):
+    """
+    Returns the start and ending index of a given
+    subword in a word.
+
+    Examples
+    ========
+    >>> from sympy.combinatorics.pc_groups import _index
+    >>> from sympy.combinatorics.free_groups import free_group
+    >>> F, x1, x2 = free_group("x1, x2")
+    >>> word = x2**2*x1**7
+    >>> w = x2**2*x1
+    >>> _index(word, w)
+    (0, 3)
+    >>> w = x1**7
+    >>> _index(word, w)
+    (2, 9)
+
+    """
+    low = -1
+    high = -1
+    for i in range(len(word)-len(w)+1):
+        if word.subword(i, i+len(w)) == w:
+            low = i
+            high = i+len(w)
+            break
+    if low == high == -1:
+        raise ValueError("Given word is not a subword")
+    return low, high
+
+def collected_word(pc_relators, word, relative_order):
+    """
+
+    """
+    group = word.group
+    uncollected_subwords = minimal_uncollected_subwords(word, relative_order)
+    power_relators, conjugate_relators = _relations(pc_relators)
+    for w, case in uncollected_subwords.items():
```

Comment:
@jksuom here after computing with `minimal_uncollected_subwords` the type of words has been changed to `core.mul`.

Reply:
It looks like `collected_subwords` is making use of array form. Perhaps the results should be converted back to group elements.

Reply:
I've tried but it doesn't work!

Reply:
The `rewritingsystem` module works with similar transformations converting words to group elements.